### PR TITLE
Support using ipv6 in make_client/make_server method

### PR DESCRIPTION
**Pull request by cocolato**
Fix: https://github.com/Thriftpy/thriftpy2/issues/204

### CODE REVIEW

**Review by ethe — APPROVED** [review]

File: thriftpy2/contrib/aio/rpc.py [hunk 1]
```diff
@@ -56,18 +59,22 @@ def make_server(service, handler,
...
-                keyfile=None, ssl_context=None, loop=None):
+                keyfile=None, ssl_context=None, loop=None,
+                socket_family=socket.AF_INET):
     processor = TAsyncProcessor(service, handler)
 
     if unix_socket:
-        server_socket = TAsyncServerSocket(unix_socket=unix_socket)
+        server_socket = TAsyncServerSocket(
+            unix_socket=unix_socket,
+            socket_family=socket_family)
```
Comment (aisk) [inline thriftpy2/contrib/aio/rpc.py:69]: If I remember correctly, `AF_INET` is the same level as `AF_UNIX`, which means we shouldn't specify the `socket_family` to `AF_INET` for unix domain socket.

**Review by aisk — CHANGES_REQUESTED** [review]

File: thriftpy2/contrib/aio/rpc.py (hunk 1 above)
Comment (cocolato) [reply thriftpy2/contrib/aio/rpc.py:80]: Thanks, updated.

**Comment by aisk** [comment]
There are still some codes where `socket_family` is applied to unix domain sockets.

**Comment by cocolato** [comment]
> There are still some codes where `socket_family` is applied to unix domain sockets.

Does this mean that when users pass in the `unix_socket` option, they will not be supported to pass in the additional `socket_family` option?

**Comment by aisk** [comment]
Yes, the unix domain socket's `socket_family` must be `AF_UNIX`. See: https://www.ibm.com/docs/en/i/7.3?topic=characteristics-socket-address-family

File: thriftpy2/contrib/aio/rpc.py [hunk 2]
```diff
@@ -30,22 +31,34 @@ async def make_client(service, host='localhost', port=9090, unix_socket=None,
...
-        socket = TAsyncSocket(unix_socket=unix_socket,
-                              connect_timeout=connect_timeout,
-                              socket_timeout=timeout)
+        socket_family = socket.AF_UNIX if not socket_family else socket_family
+        client_socket = TAsyncSocket(
+            unix_socket=unix_socket,
+            connect_timeout=connect_timeout,
+            socket_timeout=timeout,
+            socket_family=socket_family,
+        )
```
Comment (cocolato) [inline thriftpy2/contrib/aio/rpc.py:40]: Ok, I think I should not pass the socket_family option when using unix_socket.

```suggestion
        client_socket = TAsyncSocket(
            unix_socket=unix_socket,
            connect_timeout=connect_timeout,
            socket_timeout=timeout
        )
```

Comment (cocolato) [reply thriftpy2/contrib/aio/rpc.py:40]: I will rewrite this part later today.

Comment (aisk) [reply thriftpy2/contrib/aio/rpc.py:40]: Exactly

**Comment by cocolato** [comment]
> Yes, the unix domain socket's `socket_family` must be `AF_UNIX`. See: https://www.ibm.com/docs/en/i/7.3?topic=characteristics-socket-address-family

Updated.

**Pull request merged**